src/transformers/models/vitdet/modeling_vitdet.py

```diff
@@ -441,8 +441,8 @@ def window_unpartition(windows, window_size, pad_height_width, height_width):
     )
     hidden_state = hidden_state.permute(0, 1, 3, 2, 4, 5).contiguous().view(batch_size, patch_height, patch_width, -1)
 
-    if patch_height > height or patch_width > width:
-        hidden_state = hidden_state[:, :height, :width, :].contiguous()
+    # We always have height <= patch_height and width <= patch_width
```

Comment:
I think this was a victim of terrible variable naming - it should really be `padded_height` and `padded_width`. Patch height should always be less that or equal to height. I know it's not related to tracing, but could you update these for the sake of future readers?